lib/extensions/ar_slug.rb

```diff
@@ -0,0 +1,13 @@
+module ArSlug
+  extend ActiveSupport::Concern
+
+  included do
+    virtual_column :href_slug, :type => :string
```

Comment:
Interesting point, I wonder if the namespace should go beyond api as they have API's too.  Maybe we need Miq.  i.e. "miq_api_href_slug" or maybe "miq_href_slug" is sufficient in this case.